app/src/main/java/com/spascoding/taskycourse/feature_register_screen/presentation/register/RegisterViewModel.kt

```diff
@@ -39,8 +44,14 @@ class RegisterViewModel @Inject constructor(
                     email = _state.value.email,
                     password = _state.value.password,
                 ) { email, password ->
-                    authenticationUseCases.loginUser.invoke(email, password) {
-                        event.onSuccess.invoke(it.userId)
+                    GlobalScope.launch(Dispatchers.IO) {
+                        val response = authenticationUseCases.loginUser.invoke(
+                            email = email,
+                            password = password,
+                        )
+                        if (response.isSuccessful) {
+                            event.onSuccess.invoke("")
+                        }
                     }
```

Comment:
Here the issue I commented on previously can't happen since you're not executing any suspend functions after popping the back stack. Therefore, `viewModelScope` would be the better pick since generally you want to keep coroutines cancellable.